Set up InGamePanel to use CommBus

ollyau opened the pull request:
- Add reference to CommBus.js
- Register CommBusListener
- Send `SimCache.TrackerLoadedEvent` on load
- Remove `panelActive` event listener (seems to be related to visibility and/or called twice on init?)

ollyau committed:
Set up InGamePanel to use CommBus
- Add reference to CommBus.js
- Register CommBusListener
- Send SimCache.TrackerLoadedEvent on load
- Remove panelActive event listener (seems to be related to visibility and/or called twice on init?)

StevenFrost commented on Packages/PackageSources/meridian-simcache/html_ui/InGamePanels/SimCache/SimCache.js:
```diff
@@ -32,6 +35,10 @@ class SimCachePanel extends UIElement {
     }
 
+    onCommBusListenerRegistered() {
+        this.CommBusListener.callWasm("SimCache.TrackerLoadedEvent", "");
```
Is it worth adding an `onTrackerLoaded` function to contain this event broadcast for greater clarity?

ollyau replied:
I'm not entirely sure that having something called `onTrackerLoaded` would be more clear (at least to me)?  I feel like having `onTrackerLoaded` could mean several things, since there are different points in the loading process:

* The way I understand it, `connectedCallback()` is effectively the entry point (I think this just comes from having custom elements in HTML? https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements#custom_element_lifecycle_callbacks).
* In order for the CommBus to work, we need to register it, so we have `onCommBusListenerRegistered()` called _after_ that takes place.  This is the first point in the loading process where we can send the event.
* After the `SimCache.TrackerLoadedEvent` is received by the WASM module, we're planning to have the tracker receive a `SimCache.CacheDataChangedEvent` with the information about the currently tracked cache.  At that point we can actually display the current data -- which I feel concludes the loading process (since this is the first time data shows up)?

That's how I conceptualize the process, but if I see a function called `onTrackerLoaded`, I'm not sure I'd understand at what point in the loading process it's supposed to represent, just based on the name alone.  I feel like I'd _maybe_ expect it at the very end, but that wouldn't be the case, since `SimCache.TrackerLoadedEvent` needs to happen before you can reach the end.

Should I just add some comments to the code so the process is more clear?

ollyau replied:
Per discussion offline, added a new `onSubsystemsInitialized` function that's called when all subsystems are loaded. 
 (Wrapped `RegisterCommBusListener` in a `Promise` and used `Promise.all` to wait for all subsystems (currently just the CommBus) to finish loading before `onSubsystemsInitialized` is called.)

StevenFrost replied:
I think this reads much better, thanks!

ollyau committed:
Create onSubsystemsInitialized function for future extensibility
- Wrap RegisterCommBusListener callback in a Promise
- Use Promise.all to wait for all Promises to resolve
